File: Go/ExampleTests/common/orderitem_string.go
package common

type OrderItemString struct {
	Name string
	Quantity string
	Price string
	ItemTotal string
}

func NewOrderItemStringFromSlice(v []string) OrderItemString {
	s := OrderItemString{}
	if len(v) > 0 { s.Name = v[0] }
	if len(v) > 1 { s.Quantity = v[1] }
	if len(v) > 2 { s.Price = v[2] }
	if len(v) > 3 { s.ItemTotal = v[3] }
	return s
}

// NewOrderItemStringFromText builds from the text form, e.g. Money as "25 USD".
func NewOrderItemStringFromText(text string) OrderItemString {
	parts := RequireTokens(text, 4, "OrderItem")
	return OrderItemString{
		Name: parts[0],
		Quantity: parts[1],
		Price: parts[2],
		ItemTotal: parts[3],
	}
}

func (s OrderItemString) String() string {
	return Token(s.Name) + " " + Token(s.Quantity) + " " + Token(s.Price) + " " + Token(s.ItemTotal)
}

func (s OrderItemString) Equals(o OrderItemString) bool {
	return DNCEqual(s.Name, o.Name) &&
		DNCEqual(s.Quantity, o.Quantity) &&
		DNCEqual(s.Price, o.Price) &&
		DNCEqual(s.ItemTotal, o.ItemTotal)
}

func EqualOrderItemStringSlices(a, b []OrderItemString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/orderitem_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct OrderItemString {
    pub name: String,
    pub quantity: String,
    pub price: String,
    pub itemtotal: String,
}

impl OrderItemString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            name: v.get(0).copied().unwrap_or("").to_string(),
            quantity: v.get(1).copied().unwrap_or("").to_string(),
            price: v.get(2).copied().unwrap_or("").to_string(),
            itemtotal: v.get(3).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 4, "OrderItem");
        Self {
            name: parts[0].clone(),
            quantity: parts[1].clone(),
            price: parts[2].clone(),
            itemtotal: parts[3].clone(),
        }
    }
}

impl std::fmt::Display for OrderItemString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.name), crate::common::tokens::token(&self.quantity), crate::common::tokens::token(&self.price), crate::common::tokens::token(&self.itemtotal)].join(" "))
    }
}

impl PartialEq for OrderItemString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.name, &other.name)
            && crate::common::dnc_equal(&self.quantity, &other.quantity)
            && crate::common::dnc_equal(&self.price, &other.price)
            && crate::common::dnc_equal(&self.itemtotal, &other.itemtotal)
    }
}
